packages/integration-sdk-runtime/src/execution/executeIntegration.ts

```diff
@@ -218,96 +214,116 @@ export async function executeWithContext<
+          config.normalizeGraphObjectKey,
+        ),
+        graphObjectStore,
+        dataStore: new MemoryDataStore(),
+        createStepGraphObjectDataUploader,
+        beforeAddEntity: config.beforeAddEntity,
+        beforeAddRelationship: config.beforeAddRelationship,
+        afterAddEntity: createAfterAddEntityInternalHook(logger),
+        afterAddRelationship: createAfterAddRelationshipInternalHook(logger),
+        dependencyGraphOrder: config.dependencyGraphOrder,
+      });
 
-    const partialDatasets = determinePartialDatasetsFromStepExecutionResults(
-      integrationStepResults,
-    );
+      const partialDatasets = determinePartialDatasetsFromStepExecutionResults(
+        integrationStepResults,
+      );
 
-    const summary: ExecuteIntegrationResult = {
-      integrationStepResults,
-      metadata: {
-        partialDatasets,
-      },
-    };
+      const summary: ExecuteIntegrationResult = {
+        integrationStepResults,
+        metadata: {
+          partialDatasets,
+        },
+      };
 
-    await writeJsonToPath({
-      path: 'summary.json',
-      data: summary,
-    });
+      await writeJsonToPath({
+        path: 'summary.json',
+        data: summary,
+      });
 
-    context.logger.info(
-      { collectionResult: summary },
-      'Integration data collection has completed.',
-    );
-
-    processDeclaredTypesDiff(summary, (step, undeclaredTypes) => {
-      if (step.status === StepResultStatus.SUCCESS && undeclaredTypes.length) {
-        context.logger.error(
-          { undeclaredTypes },
-          `Undeclared types detected during execution. To prevent accidental data loss, please ensure that` +
-            ` all known entity and relationship types are declared.`,
-        );
-      }
-    });
+      context.logger.info(
+        { collectionResult: summary },
+        'Integration data collection has completed.',
+      );
+
+      processDeclaredTypesDiff(summary, (step, undeclaredTypes) => {
+        if (
+          step.status === StepResultStatus.SUCCESS &&
+          undeclaredTypes.length
+        ) {
+          context.logger.error(
+            { undeclaredTypes },
+            `Undeclared types detected during execution. To prevent accidental data loss, please ensure that` +
+              ` all known entity and relationship types are declared.`,
+          );
+        }
+      });
 
-    return summary;
-  } finally {
-    executionComplete = true;
+      return summary;
+    } finally {
+      executionComplete = true;
 
-    if (diskUsagePublishInterval) {
-      clearInterval(diskUsagePublishInterval);
-    }
+      if (diskUsagePublishInterval) {
+        clearInterval(diskUsagePublishInterval);
+      }
 
-    if (shouldPublishDiskUsageMetric) {
-      await tryPublishDiskUsageMetric(context);
+      if (shouldPublishDiskUsageMetric) {
+        await tryPublishDiskUsageMetric(context);
+      }
+    }
+  } finally {
+    if (config?.afterExecution) {
```

Comment:
NOTE: You should review this with "Ignore whitespace" enabled. There is very little changed in this file. I mainly wrapped this function in a `try/finally`.

<img width="674" alt="Screenshot 2023-07-05 at 2 00 17 PM" src="https://github.com/JupiterOne/sdk/assets/3771924/d169af5f-560b-4834-b393-56205cacb0fd">
e.